File: examples/app/src/main/java/com/getfastah/examples/ui/dashboard/DashboardViewModel.java
package com.getfastah.examples.ui.dashboard;

import android.app.Application;
import android.location.Location;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.getfastah.examples.LocationLiveData;
import com.getfastah.examples.NetworkLatencyLiveData;

public class DashboardViewModel extends AndroidViewModel {

    private NetworkLatencyLiveData mLatencyLive;

    public DashboardViewModel(@NonNull Application application) {
        super(application);
        mLatencyLive = new NetworkLatencyLiveData(application.getApplicationContext());
    }

    public NetworkLatencyLiveData getNetworkLatencyData() {
        return mLatencyLive;
    }
}

File: examples/app/src/main/java/com/getfastah/examples/ui/dashboard/DashboardViewModel.kt
package com.getfastah.examples.ui.dashboard

import android.app.Application
import androidx.lifecycle.*
import com.getfastah.examples.NetworkLatencyLiveData
import com.getfastah.examples.database.FastahDatabase
import com.getfastah.examples.database.MeasureSampleEntity
import com.getfastah.networkkit.MeasureSample
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext

class DashboardViewModel(application: Application, private val fastahDatabase: FastahDatabase) : AndroidViewModel(application) {

    val measureSampleHistory: LiveData<MutableList<MeasureSampleEntity>> = fastahDatabase.measureSampleDao.latestMeasurements

    class Factory(private val application: Application, private val fastahDatabase: FastahDatabase) : ViewModelProvider.Factory {
        override fun <T : ViewModel?> create(modelClass: Class<T>): T {
            return DashboardViewModel(application, fastahDatabase) as T
        }
    }
}